packages/hydrogen/src/csp/Script.tsx

```diff
@@ -0,0 +1,11 @@
+import {forwardRef} from 'react';
+import {useNonce} from './csp';
+
+type ScriptProps = JSX.IntrinsicElements['script'];
+
+export const Script = forwardRef<HTMLScriptElement, ScriptProps>(
+  (props, ref) => {
+    const nonce = useNonce();
+    return <script {...props} nonce={nonce} ref={ref} />;
+  },
+);
```

Comment:
Do we need to make this export a different name than `Script` to avoid import confusion?

Reply:
What kind of import confusion? I assume if we later have other things we do for 3p scripts, we wouldn't want multiple `Script` components.

Reply:
Along with that, perhaps we make the `CSPProvider` a more generic `HydrogenProvider`, so down the road we can add more logic to a single provider, instead of exposing many?